client/finality-grandpa/src/communication/gossip.rs

```diff
@@ -1755,17 +1802,22 @@ mod tests {
 
 		peers.new_peer(id, ObservedRole::Authority);
 
-		let mut check_update = move |update: NeighborPacket<_>| {
+		let check_update = |peers: &mut Peers<_>, update: NeighborPacket<_>| {
 			let view = peers.update_peer_state(&id, update.clone()).unwrap().unwrap();
 			assert_eq!(view.round, update.round);
 			assert_eq!(view.set_id, update.set_id);
 			assert_eq!(view.last_commit, Some(update.commit_finalized_height));
 		};
 
-		check_update(update1);
-		check_update(update2);
-		check_update(update3);
-		check_update(update4);
+		check_update(&mut peers, update1);
+		check_update(&mut peers, update2);
+		check_update(&mut peers, update3);
+		check_update(&mut peers, update4.clone());
+
+		// allow duplicate neighbor packets if enough time has passed
+		peers.inner.get_mut(&id).unwrap().view.last_update =
+			Some(Instant::now() - NEIGHBOR_REBROADCAST_AFTER);
+		check_update(&mut peers, update4);
```

Comment:
I don't like this test:

1. It modifies internal structure that can change, and the test must be updated in this case.
2. It rolls clock backwards: on systems with unsigned time representation and 0 time corresponding to the system boot, if the test is run less than two minutes after boot, it'll fail.

The alternative I see is modifying `Peers::update_peer_state()` to accept time instant as an argument, but this will require measuring time on the calling side. This seems to be even worse.

Do you see any better options?